formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/schema/ProtoBufSchemaGenerator.kt

```diff
@@ -0,0 +1,511 @@
+@file:OptIn(ExperimentalSerializationApi::class)
+
+package kotlinx.serialization.protobuf.schema
+
+import kotlinx.serialization.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.protobuf.*
+
+/**
+ * Experimental generator of ProtoBuf schema that is compatible with [serializable][Serializable] Kotlin classes
+ * and data encoded and decoded by [ProtoBuf] format.
+ *
+ * The schema is generated based on provided [SerialDescriptor] and is compatible with proto2 schema definition.
+ * An arbitrary Kotlin class represent much wider object domain than the ProtoBuf specification, thus schema generator
+ * has the following list of restrictions:
+ *
+ *  * Serial name of the class and all its fields should be a valid Proto [identifier](https://developers.google.com/protocol-buffers/docs/reference/proto2-spec)
+ *  * Nullable values are allowed only for Kotlin [nullable][SerialDescriptor.isNullable] types, but not [optional][SerialDescriptor.isElementOptional]
```

Comment:
It's the restriction of encoder, not schema generator, isn't it?

Reply:
It's still worth mentioning it here unless we have our ProtoBuf doc rephrased